src/audio/module_adapter/module_adapter.c

```diff
@@ -138,6 +138,12 @@ struct comp_dev *module_adapter_new_ext(const struct comp_driver *drv,
 		/* LL modules have the same period as the pipeline */
 		if (dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_LL)
 			dev->period = ipc_pipe->pipeline->period;
+
+		/* set dev direction based on pipeline direction */
+		if (dev->pipeline->direction_set) {
+			dev->direction_set = true;
+			dev->direction = dev->pipeline->direction;
+		}
```

Comment:
Hmm, ipc_comp_connect can propagates direction both ways, so shouldn't it be possible to set component directions based on "DAI copier"? The direction is passed to DAI copier already from host "get_gateway_direction(node_id.f.dma_type)", so direction is known.